packages/x-components/src/wiring/wires-operators.utils.ts

```diff
@@ -0,0 +1,69 @@
+import { Subject, Observable, timer, race } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { Store } from 'vuex';
+import { XBus } from '../plugins/x-bus.types';
+import { RootXStoreState } from '../store/store.types';
+import { MaybeArray } from '../utils/types';
+import { XEvent } from './events.types';
+import { TimedWireOperatorOptions, TimeSelector } from './wiring.types';
+
+/**
+ * Creates the observable for the events that will be racing the wire's execution.
+ *
+ * @param events - The events to merge its observables.
+ * @param on - The on function of the {@link XBus} where the events will run.
+ * @returns The observable for the racing events.
+ * @internal
+ */
+export function mergeEvents(events: MaybeArray<XEvent>, on: XBus['on']): Observable<void> {
+  const subject = new Subject<void>();
+  const eventsList = Array.isArray(events) ? events : [events];
+
+  /* Can't use RxJS `merge` function, as it immediately emits previously emitted values to new
+   subscriptions due to the `ReplaySubject` of the bus. With this Subject we are still receiving
+   those values immediately on subscription, but as there are no subscribers before the `return`
+   happens we are fine. */
+  eventsList.forEach(raceEvent => on(raceEvent).subscribe(() => subject.next()));
+  return subject;
```

Comment:
I was thinking on a pure rxjs solution but couldn't find one yet. However I think it should be possible...  🤔